src/net/sourceforge/kolmafia/StaticEntity.java

```diff
@@ -86,43 +83,48 @@ protected MafiaState initialValue()
 	};
 
 	public static final String getVersion()
-	{
-		return StaticEntity.getVersion( false );
-	}
-
-	public static final String getVersion( final boolean forceRevision )
 	{
 		String version = KoLConstants.VERSION_NAME;
-		if ( !KoLConstants.RELEASED || forceRevision )
+		int revision = StaticEntity.getRevision();
+		if ( revision != 0 )
 		{
-			int revision = StaticEntity.getRevision();
-			if ( revision != 0 )
-			{
-				version += " r" + revision;
-			}
+			version += " r" + revision;
 		}
 		return version;
 	}
 
 	public static final int getRevision()
 	{
-		try
+		if ( StaticEntity.cachedRevisionNumber != null )
 		{
-			if ( StaticEntity.cachedSVNRevisionNumber != null )
+			return StaticEntity.cachedRevisionNumber;
+		}
+		// Get the revision from the jar manifest attributes
+		if ( KoLConstants.REVISION == null )
+		{
+			try
 			{
-				return StaticEntity.cachedSVNRevisionNumber;
+				ClassLoader classLoader = StaticEntity.class.getClassLoader();
+				if (classLoader != null)
+				{
+					Enumeration<URL> resources = classLoader.getResources("META-INF/MANIFEST.MF");
```

Comment:
Formatting nit: missing spaces in method call.

(ditto below with manifest.getMainAttributes().getValue("Build-Revision"))